src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/StorePal.cs

```diff
@@ -44,7 +44,47 @@ public static IExportPal LinkFromCertificateCollection(X509Certificate2Collectio
 
         public static IStorePal FromSystemStore(string storeName, StoreLocation storeLocation, OpenFlags openFlags)
         {
-            throw new NotImplementedException(nameof(FromSystemStore));
+            bool isReadWrite = (openFlags & OpenFlags.ReadWrite) == OpenFlags.ReadWrite;
+            if (isReadWrite && storeLocation == StoreLocation.LocalMachine)
+            {
+                // All LocalMachine stores are read-only
+                throw new CryptographicException(
+                    SR.Cryptography_Unix_X509_MachineStoresReadOnly,
+                    new PlatformNotSupportedException(SR.Cryptography_Unix_X509_MachineStoresReadOnly));
+            }
+
+            StringComparer ordinalIgnoreCase = StringComparer.OrdinalIgnoreCase;
+            switch (storeLocation)
+            {
+                case StoreLocation.CurrentUser:
+                {
+                    // Matches Unix behaviour of getting a disallowed store that is always empty.
+                    if (ordinalIgnoreCase.Equals(X509Store.DisallowedStoreName, storeName))
+                        return new UnsupportedDisallowedStore(openFlags);
+
+                    if (ordinalIgnoreCase.Equals(X509Store.RootStoreName, storeName))
+                    {
+                        if (isReadWrite)
+                            throw new CryptographicException(SR.Security_AccessDenied);
+
+                        return new TrustedStore(storeLocation);
+                    }
```

Comment:
Can the application not add things to the AndroidCAStore?

Reply:
It can add certs through Android's Intent/Activity mechanism by kicking off an activity to request the user add the cert:
<img width="174" alt="Screen Shot 2021-03-01 at 12 40 16 PM" src="https://user-images.githubusercontent.com/47805090/109560845-33a17d00-7a91-11eb-980e-25c6befdbfaf.png">

I haven't figured out how to properly deal with the async/callback nature of activities and their results through JNI though, so I have it read-only (at least for now).

Reply:
Oh, actually, I just realised that this isn't possible in Android 11. They blocked being able to request this from a regular app - the user has to go through the phone's settings to manually install.